src/components/tl/ucp/tl_ucp_sendrecv.h

```diff
@@ -170,28 +170,17 @@ ucc_tl_ucp_resolve_p2p_by_va(ucc_tl_ucp_team_t *team, void *va, ucp_ep_h *ep,
             break;
         }
     }
-
-    if (ep_map->type == UCC_EP_MAP_STRIDED) {
-        peer = peer * ep_map->strided.stride + ep_map->strided.start;
-    }
-    else if (ep_map->type == UCC_EP_MAP_ARRAY) {
-        if (ep_map->array.elem_size == 4) {
-            uint32_t *index = ep_map->array.map;
-            peer            = index[peer];
-        }
-        else {
-            uint64_t *index = ep_map->array.map;
-            peer            = index[peer];
-        }
-    }
-    else if (ep_map->type == UCC_EP_MAP_CB) {
-        peer = (ucc_rank_t)ep_map->cb.cb(peer, ep_map->cb.cb_ctx);
-    }
-
+    core_rank = ucc_ep_map_eval(team->map, peer);
+    peer      = team->super.super.team
```

Comment:
minor: this check for presence of super.super.team (core team) is essentially a check for "service_team" which currently is only used for p2p. And i don't see it being used for 1-sided. so i believe  this extra conditional is not needed here, you can just add 
ucc_assert(team->super.super.team);